diesel/src/mysql/connection/stmt/iterator.rs

```diff
@@ -59,6 +62,26 @@ impl<'a> Row<Mysql> for MysqlRow<'a> {
     fn next_is_null(&self, count: usize) -> bool {
         (0..count).all(|i| self.binds.field_data(self.col_idx + i).is_none())
     }
+
+    fn column_count(&self) -> usize {
+        self.binds.len()
+    }
+
+    fn column_name(&self) -> Option<&str> {
+        let metadata = self.stmt.metadata().expect("Failed to get metadata");
+        let field = if self.col_idx == 0 {
+            metadata.fields()[0]
+        } else {
+            metadata.fields()[self.col_idx - 1]
+        };
+        unsafe {
+            Some(CStr::from_ptr(field.name).to_str().expect(
+                "Diesel assumes that your mysql database uses the \
```

Comment:
May be to use the `encoding` crate and get message from its real encoding? Or may be it is possibly specify desired encoding per-query basis in MySQL API?

Reply:
Every other impl that deals with stings and is part of the diesel mysql backend assumes the same. It makes no sense to change that in this one case. Changing all impls is clearly out of scope of this PR. If you are interested in this, go for it. I'm not willing to invest time in changing this in a foreseeable future, because that's something that has no value for me.